### PR TITLE
Update theme to fix mistake and forgotten change

### LINKED ISSUE #2041: [v14.0.54] thèmes obsolètes
image

Tous les thèmes de cette version sont "obsolètes"

### LINKED ISSUE #2042: [14.0.54] css perdu dans certains thèmes
image

thèmes concernés : 
- violet
- nougat

### PR DESCRIPTION
**Pull request by AlasDiablo**
Fix #2042 and #2041

### CODE REVIEW

File: src/api/models/themes.js [hunk 1]
```diff
@@ -132,12 +133,14 @@ const init = async () => {
...
             }
 
+            let hasIndex = false;
             let indexLocation = getThemeFile('default', 'index.ejs');
             if (themeConfig?.configuration?.files?.index) {
                 indexLocation = getThemeFile(
                     theme,
                     themeConfig.configuration.files.index,
                 );
+                hasIndex = true;
```
Comment (parmentf) [inline src/api/models/themes.js]: **thought (non-blocking):** Wouldn't it be more sure to test the existence of the file?‏

File: src/api/models/themes.js [hunk 2]
```diff
@@ -132,27 +135,41 @@ const init = async () => {
...
+            let hasIndex = false;
             let indexLocation = getThemeFile('default', 'index.ejs');
             if (themeConfig?.configuration?.files?.index) {
-                indexLocation = getThemeFile(
+                const unVerifiedIndexLocation = getThemeFile(
                     theme,
                     themeConfig.configuration.files.index,
                 );
+
+                if (fs.existsSync(unVerifiedIndexLocation)) {
```
Comment (parmentf) [inline src/api/models/themes.js]: **thought (non-blocking):** ‏Why use a sync I/O function?

This is inside an async function, why not use the async `exists` function (importing `fs/promises`) ?

Comment (AlasDiablo) [reply src/api/models/themes.js]: Because it doesn't exist as a promise variant...

Comment (AlasDiablo) [reply src/api/models/themes.js]: But I can use other functions to do a similar thing.

https://nodejs.org/docs/latest/api/fs.html#fspromisesaccesspath-mode

https://nodejs.org/docs/latest/api/fs.html#fspromisesstatpath-options

Comment (parmentf) [reply src/api/models/themes.js]: OK with `access` or `stat`.

**Review by touv — APPROVED** [review]

**Pull request merged**